Feature/API-only data register endpoint

ccjoel opened the pull request:
Adds `/indicators/register` endpoint which accepts:
a) `metadata` JSON file with dataset metadata (name, description, maintainer, etc)
b) `data` file for dataset (actual geospatial raw data of csv, xlsx, geotiff, netcdf formats)
c) `dictionary` file describing the dataset's fields (can be of csv, xlsx format)

I'm able to register datasets:
- Of any file type
- With annotation dictionary files of either CSV or XLSX format
- With optional metadata for file, or fallback (spreadsheets files will metadata to specify sheet, or we can make something happen to default to first sheet)

If the registration fails for any reason, and a dataset metadata/id has not been created, and simple error detail will be returned to the user. If the registration fails after having created the dataset metadata, and option is provided to the user to fix the errors listed and retry by sending the new dataset id as a query parameter on the next request. This was done to simplify flow and potential rollback complications.

Errors also include a link to the API route that generates and returns the dictionary annotation template.

Errors handled:
- Missing attributes on metadata file
- Wrong column names on dictionary
- Wrong or missing values on dictionary file
- If a file is missing, this is usually handler by the caller's client (eg curl will complain about a missing file if user tries to send something like `data=@i_dont_exist.xlsx`; should be the same for other clients)

Ideas:
- Add documentation to our user docs on dictionary file format (for both the UI dictionary file drop and this new API endpoint)
- Add documentation to our user docs specific to this endpoint
- I would like to link both urls on dictionary/metadata file format on any error responses on this endpoint. Future?

Screenshots:

![Screenshot from 2023-07-24 18-05-21](https://github.com/jataware/dojo/assets/1967061/d88ec8a1-f0f4-4adb-bae6-1deb5e726bb1)

GoogleSheets commented:
### Description

- Fixed CHIRTSmax data output column from rainfall to temperature
- fixed the bounding box description format in --help

ccjoel committed:
Checkpoint while switching context

ccjoel committed:
Another checkpoint for switching context- added info on excel sheet on
metadaya.json scripts/ file, as well as some comments/code to prepare for this
under indicators.py

ccjoel committed:
Merge remote-tracking branch 'origin/main' into feature/api-only-data-register-endpoint

ccjoel committed:
Registering datasets e2e including elwood, scaling, and job url to check status.

ccjoel committed:
added try-catch in order to clean up and inform on failures

ccjoel committed:
Had added exception handlers and debugging for error-handling/common edge case scenarios. Cleaning up in a bit.

ccjoel committed:
Modified job response as job details expire after 8 minutes adn we cant trust API users to call that link

ccjoel committed:
Modified docstring for full_dataset_register API endpoint.

ccjoel committed:
Merge branch 'feature/api-only-data-register-endpoint' of github.com:jataware/dojo into feature/api-only-data-register-endpoint

ccjoel committed:
Better error handling and user guidance. Improvements after feedback.

ccjoel committed:
Modified "testing"/mock files to register data using /register endpoint.

ccjoel committed:
Cleaning up repeated code.

ccjoel committed:
Slightly modified error details.

ccjoel committed:
Replaced error conditions from 200 to 400.

ccjoel committed:
Allowing dataset register endpoint to continue by adding the created resource to
the URL, not to metadata json file (which was inconvenient)

ccjoel committed:
Moved optional dataset id param to continue registration from route path param
to query string, as it establishes a more stable API as well as works with
openapi spec.

ccjoel committed:
Including dictionary template url on error messages in case failure is related
to bad dictionary file.

ccjoel committed:
Added endpoint to download metadata template, get help if calling
dataset/register endpoint as GET instead of POST.

ccjoel committed:
Added geotiff properties to dataset register metadata template.
